frontend/src/pages/stakingOverview/StakingGranter.js

```diff
@@ -0,0 +1,393 @@
+  const balances = useSelector((state) => state.bank.balances);
+  const feegrant = useSelector((state) => state.common.feegrant);
+
+  const [totalRewards, setTotalRewards] = React.useState(0);
+  const [availableBalance, setAvailableBalance] = useState(0);
+  const [stakingOpen, setStakingOpen] = React.useState(false);
+  const [undelegateOpen, setUndelegateOpen] = React.useState(false);
+  const [selectedValidator, setSelectedValidator] = useState({});
+  const [redelegateOpen, setRedelegateOpen] = React.useState(false);
+
+  const handleDialogClose = () => {
+    setStakingOpen(false);
+    setUndelegateOpen(false);
+    setRedelegateOpen(false);
+  };
+
+  const getAuthzBalances = () => {
+    dispatch(
+      getBalances({
+        baseURL: chainInfo.config.rest + "/",
+        address: granter,
+        chainID: chainID,
+      })
+    );
+  };
+
+  const fetchGranterDelegationsInfo = () => {
+    dispatch(
+      getAuthzDelegations({
+        baseURL: chainInfo?.config?.rest,
+        chainID: chainID,
+        address: granter,
+      })
+    );
+  };
+
+  const fetchAuthzDelegatorTotalRewards = () => {
+    dispatch(
+      getAuthzDelegatorTotalRewards({
+        baseURL: chainInfo?.config?.rest,
+        chainID: chainID,
+        address: granter,
+      })
+    );
+  };
+
+  const onAuthzDelegateTx = (data) => {
+    authzExecHelper(dispatch, {
+      type: "delegate",
+      address: address,
+      baseURL: chainInfo.config.rest,
+      delegator: granter,
+      validator: data.validator,
+      amount: data.amount * 10 ** currency.coinDecimals,
+      denom: currency.coinMinimalDenom,
+      chainId: chainInfo.config.chainId,
+      rest: chainInfo.config.rest,
+      aminoConfig: chainInfo.aminoConfig,
+      prefix: chainInfo.config.bech32Config.bech32PrefixAccAddr,
+      feeAmount:
+        chainInfo.config.gasPriceStep.average * 10 ** currency.coinDecimals,
+      feegranter: feegrant?.granter,
+    });
+  };
+
+  const onAuthzUndelegateTx = (data) => {
+    authzExecHelper(dispatch, {
+      type: "undelegate",
+      address: address,
+      delegator: granter,
+      validator: data.validator,
+      amount: data.amount * 10 ** currency.coinDecimals,
+      denom: currency.coinMinimalDenom,
+      chainId: chainInfo.config.chainId,
+      rest: chainInfo.config.rest,
+      aminoConfig: chainInfo.aminoConfig,
+      prefix: chainInfo.config.bech32Config.bech32PrefixAccAddr,
+      feeAmount:
+        chainInfo.config.gasPriceStep.average * 10 ** currency.coinDecimals,
+      feegranter: feegrant.granter,
+    });
+  };
+
+  const onAuthzRedelegateTx = (data) => {
+    authzExecHelper(dispatch, {
+      type: "redelegate",
+      address: address,
+      baseURL: chainInfo.config.rest,
+      delegator: granter,
+      srcVal: data.src,
+      destVal: data.dest,
+      amount: data.amount * 10 ** currency.coinDecimals,
+      denom: currency.coinMinimalDenom,
+      chainId: chainInfo.config.chainId,
```

Comment:
seem like you have called with some common fields to all the functions. so better to define some common object at the beginning of the function. use it in all places. i.e, chainId, rest,aminoConfig, feeAmount.

Reply:
Lets create a follow up issue for this. We will address this in separate PR.